### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by bernarduh** [python-jsonschema/jsonschema#961] [Q&A] [answered]
Hi guys,

I'm working on a project that uses this lib as a dependency and I'm trying to understand which license I can use. I noticed that you have the _importlib\_resources_ in your dependencies and that project uses an Apache license. Is it ok for us to use the MIT license?

**Comment by Julian** [comment, marked as answer]
Hi there. I can't give legal advice unfortunately.

If you're asking what license `jsonschema` is under, it's [MIT](https://github.com/python-jsonschema/jsonschema/blob/main/COPYING).

It has dependencies, and they indeed have their own licenses (as does Python the language itself).

But if you're asking whether that affects what license you can put on your own source code? That it should not, though again, if you're truly concerned, you'd need to speak to an IP lawyer.

**Reply by bernarduh** [reply]
Hi @Julian 

Thanks for the quick feedback.

I was under the impression that if we use a dependency with Apache we would have to use the Apache license and not MIT. Based on :point_down: 

https://ghinda.com/blog/opensource/2020/open-source-licenses-apache-mit-bsd.html#compatibilities-with-other-open-source-licenses
> MIT is a permissive license allowing you to re-license the derivative work under any license ([source](https://law.stackexchange.com/questions/6081/can-i-bundle-mit-licensed-components-in-a-apache-2-0-licensed-project)). In general, when you combine the MIT License with any other open source license, the resulting derivative work will not be MIT License. When you combine Public Domain, or MIT Licensed components, the result derivative work can be MIT License.

I'm trying to understand if you picked the MIT license based on some legal reasoning because _jsonschema_ is the only dependency in my project that traces down to an Apache license (_jsonschema_ -> _importlib-resource_) and if you're under infringement, them I'm liable for that infringement too.

Thanks :slightly_smiling_face:

**Reply by Julian** [reply]
There is no license infringement happening in this repository.

**Reply by bernarduh** [reply]
Ok.Thank you.
